### PR TITLE
Winduntoo support.

### PR DESCRIPTION
**Pull request by nicolasnoble**
Things to consider to make grpc binaries to run under Windows' Ubuntu layer:

-) epoll_create1 and epoll_wait work under that layer, but not epoll_pwait.
-) TCP_NODELAY can't be set.

### CODE REVIEW

**Comment by nicolasnoble** [comment]
I realize this is probably leaking memory.

File: src/core/lib/iomgr/tcp_client_posix.c [hunk 1]
```diff
@@ -86,8 +86,9 @@ static grpc_error *prepare_socket(const grpc_resolved_address *addr, int fd,
   err = grpc_set_socket_cloexec(fd, 1);
   if (err != GRPC_ERROR_NONE) goto error;
   if (!grpc_is_unix_socket(addr)) {
-    err = grpc_set_socket_low_latency(fd, 1);
-    if (err != GRPC_ERROR_NONE) goto error;
+    /* We're going to ignore the potential error on this one, because it's not
+       a fatal problem per-se. */
+    grpc_set_socket_low_latency(fd, 1);
```
Comment (nicolasnoble) [inline src/core/lib/iomgr/tcp_client_posix.c]: The environment behaves really strongly the same as any other Linux. The only difference I've managed to spot is the result of uname.

Comment (nicolasnoble) [inline src/core/lib/iomgr/tcp_client_posix.c]: Microsoft/BashOnWindows#423

Comment (ctiller) [inline src/core/lib/iomgr/tcp_client_posix.c]: So we could have a is_winduntu() { strstr(fread("/proc/version"), "Microsoft") != 0 }

**Comment by ctiller** [comment]
Closing since this seems stale

**Pull request closed without merging**